pkg/gpu/stats.go

```diff
@@ -51,14 +54,15 @@ func (g *statsGenerator) getStats(nowKtime int64) *model.GPUStats {
 		}
 	}
 
-	g.configureNormalizationFactor()
+	normFactor := g.getNormalizationFactor()
 
 	stats := model.GPUStats{
-		PIDStats: make(map[uint32]model.PIDStats),
+		ProcessStats: make(map[uint32]model.PIDStats),
 	}
 
-	for pid, aggregator := range g.aggregators {
-		stats.PIDStats[pid] = aggregator.getStats()
+	for pid, aggr := range g.aggregators {
+		aggr.setGPUUtilizationNormalizationFactor(normFactor)
```

Comment:
why do you need to have it as a struct field? this is a global value for all the aggregators, and will be the same. just pass it to the `getStats`

Reply:
I had it as a struct field as for me the aggregation struct has the struct fields as input, and then the function doesn't receive arguments, which looks cleaner to me as there's no mix of which arguments are passed as function arguments and which as struct attributes. But it's a small thing, so I can change it no problem.

Reply:
struct fields are "allocating" more memory, as they have the object lifecycle, while the function parameter is cleaned once the function is finished. in this case i don't think this parameter is actually needed by the object in multiple places, and instead the only usage is actually scoped to just one function (`getStats`), hence i proposed to pass it as a parameter

Reply:
Changed!